optuna/_gp/_search_space.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+import math
+from typing import Any
+from typing import NamedTuple
+
+import numpy as np
+import scipy.stats.qmc
+
+import optuna
+
+
+LINEAR = 0
+LOG = 1
+CATEGORICAL = 2
+
+
+class SearchSpace(NamedTuple):
```

Comment:
Sorry, it is an side effect of trying to use numba (numba only supports `NamedTuple` and not `dataclass`, cf. https://github.com/numba/numba/issues/4037). I'll change to `dataclass` for now since we don't need `numba` now.